lib/event_store/event_store.ex

```diff
@@ -1,7 +1,39 @@
 defmodule Shared.EventStore do
-  defmacro __using__(_opts \\ []) do
-    quote location: :keep, generated: true do
+  defmodule Util do
```

Comment:
What is the reasoning to make these functions public and wrap them into its own module? Wouldn't it suffice to declare these functions as `defp`s?

Reply:
I've changed the implementation so the name makes sense now. In every phoenix app, you list all of the available ecto repositories under the key `:ecto_repos`, e.g. `config :rechnungsstellung, ecto_repos: [Rechnungsstellung.Repo]`.
I take this `Rechnungsstellung.Repo` if it is the only repository configured. So it is the _default_ repo for this application actually

Reply:
> What is the reasoning to make these functions public and wrap them into its own module? Wouldn't it suffice to declare these functions as `defp`s?

My intention was to reduce the amount of code generated. Since our default setup consists of just one event store, it's fine to declare them as private inside the module.
The actual reason I've put them into a new module was my incompetence :-). The following code produced an error I couldn't fix.
```elixir
@repository Keyword.get(opts, :repo) || case(Application.get_env(@otp_app, :ecto_repos)) do
                       [repo] ->
                         repo

                       _ ->
                         IO.warn(
                           ":repo option required if you want to wrap append_event in a transaction."
                         )
                     end
```
Brackets around the `case` block fixed it...